accern_xyme/accern_xyme.py

```diff
@@ -741,6 +741,78 @@ def get_settings(self) -> Dict[str, Any]:
         assert self._settings is not None
         return self._settings
 
+    def get_timing(self, waiting: str) -> Dict[str, Any]:
+
+        nodes = self.get_nodes()
+        node_list = []
+        node_total = []
+        node_avg = []
+        node_name: str = []
+        dict_temp = {}
+
+        def del_waiting(node_time: list) -> List[Any]:
+            for pos, cur in enumerate(node_time):
+                for key, value in cur.copy().items():
+                    if value == waiting:
+                        del node_time[pos]
+            return node_time
+
+        for node_ix, node in enumerate(nodes):
+            node_list.append(node)
+            node_time = self.get_node(nodes[node_ix]).get_timing()
+            node_time = del_waiting(node_time)
+
+            dicts: Dict[str, Any] = {nodes[node_ix]: {}}
+
+            total_sum = 0
+            for pos, cur in enumerate(node_time):
+                length = len(node_time)
+                dicts[nodes[node_ix]].setdefault("fns", []).append(cur)
+
+                for key, value in enumerate(node_time[pos]):
```

Comment:
enumerate gives you the position and the element. key will never be a string in this case